src/tests/Microsoft.PowerFx.Interpreter.Tests/MutationFunctionsTests.cs

```diff
@@ -224,6 +225,24 @@ public void MutationCheckFailTests(string expression)
             }
         }
 
+        // Immutable arguments
+        [Theory]
+        [InlineData("Patch([1,2,3], {Value:1}, {Value:9})")]
+        [InlineData("With({x:[1,2,3]},Patch(x, {Value:1}, {Value:9}))")]
+        [InlineData("Patch(namedFormula, {Value:1}, {Value:9})")]
+
+        [InlineData("Collect([1,2,3], {Value:9})")]
+        [InlineData("With({x:[1,2,3]},Collect(x, {Value:9}))")]
```

Comment:
> [InlineData("With({x:[1,2,3]},Collect(x, {Value:9}))")] [](http://example.com/codeflow?start=7&length=56)

beware - this works in power apps today

Reply:
Yep, which is why the new mutation bit is only checked if we are using PowerFxV1CompatibilityRules

Reply:
What is the alternative expression we'd recommend they write? Eg, do they need a Set in there or something?